modules/instance/main.tf

```diff
@@ -1,46 +1,117 @@
+# ##############################################################################
+# # Create Virtual Network Interface
+# ##############################################################################
+
+resource "ibm_is_virtual_network_interface" "primary_vni" {
+  count                     = length(var.primary_network_interface) > 0 && !var.use_legacy_network_interface ? length(var.primary_network_interface) : 0
+  subnet                    = var.primary_network_interface[count.index].subnet
+  name                      = (var.primary_network_interface[count.index].interface_name != "" ? var.primary_network_interface[count.index].interface_name : null)
+  security_groups           = (var.primary_network_interface[count.index].security_groups != null ? var.primary_network_interface[count.index].security_groups : [])
+  allow_ip_spoofing         = var.primary_network_interface[count.index].allow_ip_spoofing
+  auto_delete               = false
+  enable_infrastructure_nat = true
+  dynamic "primary_ip" {
+    for_each = var.primary_network_interface[count.index].primary_ipv4_address != null ? [1] : []
+    content {
+      address = var.primary_network_interface[count.index].primary_ipv4_address
+    }
+  }
+}
+
+resource "ibm_is_virtual_network_interface" "additional_vnis" {
+  count                     = length(var.network_interfaces) > 0 && !var.use_legacy_network_interface ? length(var.network_interfaces) : 0
+  name                      = var.network_interfaces[count.index].interface_name
+  subnet                    = var.network_interfaces[count.index].subnet
+  allow_ip_spoofing         = var.network_interfaces[count.index].allow_ip_spoofing
```

Comment:
`allow_ip_spoofing` is optional. This piece of code migh fail without any validation around it

Reply:
Validation not required. I have tested it, if no value is set it is set to false.